gnucash/report/standard-reports/net-charts.scm

```diff
@@ -263,6 +263,61 @@
     (define (monetary->double monetary)
       (gnc-numeric-to-double (gnc:gnc-monetary-amount monetary)))
 
+    (define (split->date s)
+      (xaccTransGetDate (xaccSplitGetParent s)))
+
+    ;; this function will scan through the account splitlist, building
+    ;; a list of balances along the way. it will use the dates
+    ;; specified in the variable dates-list.
+    ;; input: account
+    ;;  uses: dates-list (list of time64)
+    ;;   out: (list account bal0 bal1 ...)
+    (define (account->balancelist account)
+      (let loop ((splits (stable-sort! (xaccAccountGetSplitList account)
+                                       (lambda (a b)
+                                         (< (split->date a) (split->date b)))))
```

Comment:
@jralls from experimentation, it would seem that xaccAccountGetSplitList will *always* return a splitlist, sorted by posted-date. If this is correct, I can remove sorting here which will be rather nice. Can you confirm if the engine will enforce this assumption? Reviewing xaccAccountGetBalanceAsOfDate will call sorting but states it would usually be a NOOP. https://github.com/Gnucash/gnucash/blob/maint/libgnucash/engine/Account.cpp#L3293

Reply:
@christopherlam You actually wanted L3701, but yes, it calls xaccAccountSortSplits to make sure that they're in the correct order. That's a bit more complicated than posted date because it has more sorts when there are more than one split posted on a particular date, but date is the outermost order so sorting the returned list by date is a no-op.

Reply:
Happy times!